src/EnergyPlus/DXCoils.cc

```diff
@@ -9992,6 +9993,110 @@ Label50: ;
+		Real64 SHR( 0.0 ); // the result - the adjusted design SHR
+
+		// Locals
+		// FUNCTION ARGUMENT DEFINITIONS:
+
+		// FUNCTION PARAMETER DEFINITIONS:
+		// na
+
+		// FUNCTION LOCAL VARIABLE DECLARATIONS:
+		Real64 CBF_calculated( 0.0 ); // coil bypass factor based on TotCap, AirFlow, SHR, and BaroPress
+		Real64 InletAirEnthalpy( 0.0 ); // Enthalpy of inlet air to evaporator at given conditions [J/kg]
+		Real64 DeltaH( 0.0 ); // Enthalpy drop across evaporator at given conditions [J/kg]
+		Real64 HTinHumRatOut( 0.0 ); // Air enthalpy at inlet air temp and outlet air humidity ratio [J/kg]
+		Real64 DeltaHumRat( 0.0 ); // Humidity ratio drop across evaporator at given conditions [kg/kg]
+		Real64 OutletAirTemp( 0.0 ); // Outlet dry-bulb temperature from evaporator at given conditions [C]
+		Real64 OutletAirEnthalpy( 0.0 ); // Enthalpy of outlet air at given conditions [J/kg]
+		Real64 OutletAirHumRat( 0.0 ); // Outlet humidity ratio from evaporator at given conditions [kg/kg]
+		Real64 OutletAirRH( 0.0 ); // relative humidity of the outlet air
+		Real64 CalcADPTemp( 0.0 ); // actual ADP temperature based on bypass factor [C]
+		Real64 CalcADPHumRat( 0.0 ); // actual ADP humidity ratio based on bypass factor [kg/kg]
+		Real64 CalcADPTempFnHR( 0.0 ); // actual ADP temperature as a function of humidity ratio based on bypass factor [C]
+		Real64 ADPerror( 0.0 ); // difference between ADP function of temperature and humidity ratio [deltaC]
+		bool bStillValidating( true ); // while loop flag
+		bool bNoReporting( false ); // don't report specific warnings in calcCBF while iterating on result
+		bool bReversePerturb( false ); // identifies when SHR is being lowered based on outlet air RH
+
+		SHR = InitialSHR;
+		while ( bStillValidating ) {
+			CBF_calculated = max( 0.0, CalcCBF( UnitType, UnitName, RatedInletAirTemp, RatedInletAirHumRat, TotCap, AirMassFlow, SHR, bNoReporting ) );
+			DeltaH = TotCap / AirMassFlow;
+			InletAirEnthalpy = PsyHFnTdbW( RatedInletAirTemp, RatedInletAirHumRat );
+			HTinHumRatOut = InletAirEnthalpy - ( 1.0 - SHR ) * DeltaH;
+			OutletAirHumRat = PsyWFnTdbH( RatedInletAirTemp, HTinHumRatOut, CallingRoutine );
+			DeltaHumRat = RatedInletAirHumRat - OutletAirHumRat;
+			OutletAirEnthalpy = InletAirEnthalpy - DeltaH;
+			OutletAirTemp = PsyTdbFnHW( OutletAirEnthalpy, OutletAirHumRat );
+			OutletAirRH = PsyRhFnTdbWPb( OutletAirTemp, OutletAirHumRat, StdBaroPress, CallingRoutine );
+			if ( CBF_calculated < 1 ) {
+				CalcADPTemp = RatedInletAirTemp - ( ( RatedInletAirTemp - OutletAirTemp ) / ( 1 - CBF_calculated ) );
+				CalcADPHumRat = RatedInletAirHumRat - ( ( DeltaHumRat ) / ( 1 - CBF_calculated ) );
+				CalcADPTempFnHR = PsyTdpFnWPb( CalcADPHumRat, StdBaroPress, CallingRoutine );
+				ADPerror = CalcADPTemp - CalcADPTempFnHR;
+			} else {
+				ADPerror = 0; // might be able to check for RH >= 1 and reduce SHR, need defect file for that since can't create one
+			}
+
+			if ( std::abs( ADPerror )  > 0.012 ) {
```

Comment:
To finish this thought, the actual solution has 2 unknowns, ADP and CBF. This fix only picks the first available solution.